src/CabanaPD_DisplacementProfile.hpp

```diff
@@ -0,0 +1,108 @@
+/****************************************************************************
+ * Copyright (c) 2022-2023 by Oak Ridge National Laboratory                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of CabanaPD. CabanaPD is distributed under a           *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef DISPLACEMENTPROFILE_H
+#define DISPLACEMENTPROFILE_H
+
+#include <Kokkos_Core.hpp>
+
+#include <Cabana_Core.hpp>
+
+namespace CabanaPD
+{
+
+auto getDim( const int dim )
+{
+    Kokkos::Array<int, 2> orthogonal;
+    orthogonal[0] = ( dim + 1 ) % 3;
+    orthogonal[1] = ( dim + 2 ) % 3;
+    return orthogonal;
+}
+
+template <typename ParticleType, typename UserFunctor>
+void createDisplacementProfile( MPI_Comm comm, const int num_cell,
+                                const int profile_dim, std::string file_name,
+                                ParticleType particles, UserFunctor user )
+{
+    using memory_space = typename ParticleType::memory_space;
+    auto profile = Kokkos::View<double* [3], memory_space>(
```

Comment:
@streeve I noted in the Elastic Wave example we have "auto profile = Kokkos::View<double* [2], memory_space>", i.e., "[2]" instead of "[3]". Why the difference?

Reply:
Copy paste mistake - fixed. You added the magnitude in the same View, but I've kept them separate for reusability